### PR TITLE
Getting element access to break more consistent in long invocation chains.

### LINKED ISSUE #956: Seems like incorrect break for chained property and method calls
In https://playground.csharpier.com/ I have played around with chains and following does not seem right:
Input:
```
someLongNameField
    .Method0()
    .Property0
    .Property1
    .Property2
    .Method1()
    .Property3
    .Array1[1]
    .Property4
    .Property5
    .Method2()
    .Method3("some input")
    .Method2()
    .Property6
    .Property7
    .Array2[2]
    .Property8
    .Method4("some input")
    .Property9
    .Property10;
```

output:
```
someLongNameField.Method0().Property0.Property1.Property2.Method1().Property3.Array1[
    1
].Property4.Property5
    .Method2()
    .Method3("some input")
    .Method2()
    .Property6.Property7.Array2[2].Property8
    .Method4("some input")
    .Property9.Property10;
```
property and method chains are broken differently.

Other example:
input
```
			return _someResolver
				.Resolve(
					(
						raw
							.Settings
							.AttributeTemplate
							.Settings
							.OutputSettings[raw.TestFramework]
							.AttributeTemplate
							.Settings
							.AttributeTemplate
							.Settings
							.OutputSettings[raw.TestFramework]
							.AttributeTemplate,
						new
						{
							ClassName = className,
							TestClassName = raw.MethodSymbol.ContainingType.Name,
							TestName = raw.MethodSymbol.Name
						})
				);
```

output:
```
return _someResolver.Resolve(
    (
        raw.Settings.AttributeTemplate.Settings.OutputSettings[raw.TestFramework]
            .AttributeTemplate
            .Settings
            .AttributeTemplate
            .Settings
            .OutputSettings[raw.TestFramework].AttributeTemplate,
        new
        {
            ClassName = className,
            TestClassName = raw.MethodSymbol.ContainingType.Name,
            TestName = raw.MethodSymbol.Name
        }
    )
);
```

### PR DESCRIPTION
**Pull request by belav**
closes #956

### CODE REVIEW

**Comment by belav** [comment]
This had a number of side effects I need to review. At first glance some of these changes are for the better, but some we maybe don't want.

https://github.com/belav/csharpier-repos/pull/89/files

**Comment by belav** [comment]
This morphed a bit, and turned into getting properties to break more consistently with invocations and array access.
I believe there are still a couple things to improve, but this does resolve some of the issues in #451

**Comment by shocklateboy92** [comment]
So uh... is this ready for review now?

File: Src/CSharpier.Tests/FormattingTests/TestFiles/cs/MemberChain_PropertiesConsistent.test [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    .CallMethod();
+
+var someVariable = someObject
+    .Property()
+    .CallMethod(someValue => someValue.SomeProperty == someOtherValue___________________________)
+    .CallMethod();
+
+// TODO too hard to change this for now, will do it in https://github.com/belav/csharpier/issues/451
+var someVariable = this.Property.CallMethod(
+    someValue => someValue.SomeProperty == someOtherValue___________________________
```
Comment (shocklateboy92) [inline Src/CSharpier.Tests/FormattingTests/TestFiles/cs/MemberChain_PropertiesConsistent.test:21]: also this isn't all that bad.

**Review by shocklateboy92 — APPROVED** [review]
LGTM.

**Pull request merged**